test/github/graphql.test.js

```diff
@@ -0,0 +1,80 @@
+const GitHub = require('../../lib/github')
+const nock = require('nock')
+const Bottleneck = require('bottleneck')
+
+describe('github/graphql', () => {
+  let github
+
+  // Expect there are no more pending nock requests
+  beforeEach(async () => nock.cleanAll())
+  afterEach(() => expect(nock.pendingMocks()).toEqual([]))
+
+  beforeEach(() => {
+    const logger = {
+      debug: jest.fn(),
+      trace: jest.fn()
+    }
+
+    // Set a shorter limiter, otherwise tests are _slow_
+    const limiter = new Bottleneck({ maxConcurrent: 1, minTime: 1 })
+
+    github = new GitHub({ logger, limiter })
+  })
+
+  describe('query', () => {
+    let query
+    let data
+
+    test('makes a graphql query', async () => {
+      query = 'query { viewer { login } }'
+      data = { viewer: { login: 'bkeepers' } }
+
+      nock('https://api.github.com', {
+        reqheaders: { 'content-type': 'application/json' }
+      }).post('/graphql', {query})
+        .reply(200, { data })
+
+      expect(await github.query(query)).toEqual(data)
+    })
+
+    test('makes a graphql query with variables', async () => {
+      const variables = {owner: 'probot', repo: 'test'}
+
+      nock('https://api.github.com', {
+        reqheaders: { 'content-type': 'application/json' }
+      }).post('/graphql', {query, variables})
```

Comment:
What is `query` here? Does that come from the `nock` library?

Reply:
It's defined above to be:

```js
query = 'query { viewer { login } }'
```

This is how nock has you assert request parameters that are sent 😞

Reply:
> It's defined above to be:…

Lol, now it is 98e2e15fecf6ea58177528c822b7a817022ed08a